hostinfo/base.lua

```diff
@@ -97,16 +97,20 @@ function HostInfo:_pushError(err)
 end
 
 function HostInfo:_pushParams(err, data)
-    -- flatten single entry objects
-    if type(data) == 'table' then
-      if #data == 1 then data = data[1] end
-    end
-    self._params = data
-    if err then
+  -- flatten single entry objects
+  if type(data) == 'table' then
+    if #data == 1 then data = data[1] end
+  end
+  self._params = data
+  if err then
+    if type(err) == 'table' and next(err) then
       self:_pushError(err)
-    elseif not err and not data or not next(data) then
+    elseif type(err) == 'string' and #err > 0 then
       self:_pushError(err)
     end
+  elseif not err and not data or not next(data) then
```

Comment:
Yes, this is by design to catch hostinfos that should return data but don't. 
E.g. We have a hostinfo that on the host system instead of returning data errors out but is lacking proper error handling code, the returned output will say something along the lines of:
```
{
  metrics: {}
  error: "No error specified but no data recieved"
}
```
So if a hostinfo isnt going to return data it should ideally tell you something along the lines of command not found as an error message and return out, 
HIs that need to return empty metrics by design. e.g. cron, dont use the _pushParams convenience method.